server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -914,8 +914,26 @@ public void insert(InsertPlan insertPlan) throws QueryProcessException {
       insertPlan.setSchemasAndTransferType(schemas);
       StorageEngine.getInstance().insert(insertPlan);
       if (insertPlan.getFailedMeasurements() != null) {
-        throw new StorageEngineException(
-            "failed to insert points " + insertPlan.getFailedMeasurements());
+        // check if all path not exist exceptions
+        List<String> failedPaths = new ArrayList<>(insertPlan.getFailedMeasurements().keySet());
+        List<Exception> exceptions = new ArrayList<>(insertPlan.getFailedMeasurements().values());
+        boolean isPathNotExistException = true;
+        for(Exception e : exceptions){
+          Exception curException = e;
+          while(curException.getCause() != null){
+            curException = (Exception) curException.getCause();
```

Comment:
You may just define this as a Throwable so you may avoid the typecast.

Reply:
Thanks for yoru suggestion.